analyzers/src/SonarAnalyzer.CSharp/Rules/InfiniteRecursion.RoslynCfg.cs

```diff
@@ -0,0 +1,151 @@
+        {
+            public void CheckForNoExitProperty(SyntaxNodeAnalysisContext c, PropertyDeclarationSyntax property, IPropertySymbol propertySymbol)
+            {
+                if (property.ExpressionBody?.Expression != null)
+                {
+                    var cfg = ControlFlowGraph.Create(property.ExpressionBody, c.SemanticModel);
+                    var walker = new RecursionSearcher(new RecursionContext<ControlFlowGraph>(cfg, propertySymbol, property.Identifier.GetLocation(), c, "property's recursion"));
+                    walker.CheckPaths();
+                }
+                else if (property.AccessorList != null)
+                {
+                    foreach (var accessor in property.AccessorList.Accessors.Where(a => a.HasBodyOrExpressionBody()))
+                    {
+                        var cfg = ControlFlowGraph.Create(accessor, c.SemanticModel);
+                        var context = new RecursionContext<ControlFlowGraph>(cfg, propertySymbol, accessor.Keyword.GetLocation(), c, "property accessor's recursion");
+                        var walker = new RecursionSearcher(context, !accessor.Keyword.IsKind(SyntaxKind.SetKeyword));
+                        walker.CheckPaths();
+                    }
+                }
+            }
+
+            public void CheckForNoExitMethod(SyntaxNodeAnalysisContext c, CSharpSyntaxNode body, SyntaxToken identifier, ISymbol symbol)
+            {
+                var operation = c.SemanticModel.GetOperation(body.Parent);
+                var cfg = ControlFlowGraph.Create(operation.RootOperation().Syntax, c.SemanticModel);
+                var enclosingKinds = new HashSet<SyntaxKind>
+                {
+                    SyntaxKindEx.LocalFunctionStatement, SyntaxKind.SimpleLambdaExpression, SyntaxKind.AnonymousMethodExpression, SyntaxKind.ParenthesizedLambdaExpression
+                };
+                if (c.Node.IsKind(SyntaxKindEx.LocalFunctionStatement))
+                {
+                    // we need to go up and track all possible enclosing local function statements
+                    foreach (var enclosingFunction in c.Node.Ancestors().Where(x => enclosingKinds.Contains(x.Kind())).Reverse())
+                    {
+                        if (enclosingFunction.IsKind(SyntaxKindEx.LocalFunctionStatement))
+                        {
+                            cfg = cfg.GetLocalFunctionControlFlowGraph(c.SemanticModel.GetDeclaredSymbol(enclosingFunction) as IMethodSymbol);
+                        }
+                        else
+                        {
+                            var operationWrapper = cfg.FlowAnonymousFunctionOperations().Single(x => x.WrappedOperation.Syntax == enclosingFunction);
+                            cfg = cfg.GetAnonymousFunctionControlFlowGraph(operationWrapper);
+                        }
+                        if (cfg == null)
+                        {
+                            return;
+                        }
```

Comment:
The null should not happen indeed, we can remove it. See
https://github.com/dotnet/roslyn/blob/6a9d04542e4df19817e2526d1f68ec7687f93ce9/src/Compilers/Core/Portable/Operations/ControlFlowGraph.cs#L285-L288

https://github.com/dotnet/roslyn/blob/6a9d04542e4df19817e2526d1f68ec7687f93ce9/src/Compilers/Core/Portable/Operations/ControlFlowGraph.cs#L236-L239